app/api/server/lib/rooms.js

```diff
@@ -110,11 +107,12 @@ export async function findChannelAndPrivateAutocomplete({ uid, selector }) {
 			name: 1,
 		},
 	};
-	const userRooms = Subscriptions.cachedFindByUserId(uid, { fields: { rid: 1 } })
+
+	const userRoomsIds = Subscriptions.cachedFindByUserId(uid, { fields: { rid: 1 } })
 		.fetch()
 		.map((item) => item.rid);
 
-	const rooms = await Rooms.findChannelAndPrivateByNameStarting(selector.name, userRooms, options).toArray();
+	const rooms = await Rooms.findRoomsWithoutDiscussionsByRoomIds(selector.name, userRoomsIds, options).toArray();
```

Comment:
This will affect every room auto complete field on rocket.chat, not only the one in the create discussion modal.
I don't know if that's ok or not.

Reply:
I think that's ok considering that currently we filter rooms in autocomplete by the `name` field, BUT discussions' "real names" are stored in the `fname` field (`name` only stores a code in discussions).
So I guess that currently autocomplete is not efficient to find discussions (which means this change would just remove these discussions' codes from the autocomplete). What do you think?